networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,100 @@
+import networkx as nx
+from networkx import NetworkXError
+from networkx.utils import not_implemented_for
+
+__all__ = [
+    "tree_broadcast_time",
+]
+
+
+def _get_max_broadcast_value(G, U, v, values):
+    adj = sorted(set(G.neighbors(v)) & U, key=lambda u: values[u], reverse=True)
+    return max(values[u] + i for i, u in enumerate(adj, start=1))
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def tree_broadcast_time(G):
+    """Return the optimal Broadcast Time of the tree G.
```

Comment:
I agree that the terminology used here is a bit confusing, this is definitely something that I am still trying to completely understand as of now. 
I think "broadcast time" is typically referring to the broadcast time of a graph (which in turn represents the max broadcast time from all its originators).
I decided to remove any mention of "optimal broadcast time" as it essentially means the same thing as "minimum broadcast time". In this case, "minimum broadcast time" refers to trying to find the shortest "broadcast time" for the graph, so I don't think we should allow users to specify a specific starting node. 
I propose two steps to deal with these differences:
1. We can rename the current function to ``min_tree_broadcast_time``
2. We can make a new function that returns the ``tree_broadcast_time``, which would simply be the largest ``distance`` from a vertex to the ``broadcast_center`` (which is the optimum starting point for the "minimum broadcast time") plus the value of ``min_tree_broadcast_time``.

Reply:
This problem is also different from minimum broadcast time from an originator (which would require a fixed starting node like you suggested, or random if unspecified by the user). I am thinking we could call that something like ``min_broadcast_scheme``. I feel like finding good function names is challenging because of these similarities in "broadcast time" of a vertex vs "broadcast time" of a graph. 
Maybe the following notation could help us?
The minimum number of time steps required to broadcast from v to all vertices in the graph G is called the broadcast time of vertex v, denoted b(G;v). The maximum broadcast time over all originator vertices in G is called the broadcast time of G, denoted b(G) (Harutyunyan1 and Li 2019).

Reply:
>The minimum number of time steps required to broadcast from v to all vertices in the graph G is called the broadcast time of vertex v, denoted b(G;v). The maximum broadcast time over all originator vertices in G is called the broadcast time of G, denoted b(G) (Harutyunyan1 and Li 2019).

This sounds like a good definition to me. We could then have one function `broadcast_time(G, node=None, ...)` which finds the "broadcast time of a node in G" `b(G; node)` and if `node is None` then it returns the "broadcast time of the graph G" `b(G)`.

Thoughts?

Reply:
Yea that's perfect I think. I am currently working on updates to the PR (and many more tests!). 
We might need to go through a second round of reviews as I need to introduce another concept called the broadcast centers.